### ISSUE TITLE
Inconsistent loading of environment variables between prisma cli (npx prisma) and prisma client

### ISSUE THREAD

**Issue by arunlodhi** [prisma/prisma#16890]
### Bug description

I am using Prisma within a NextJS app. My DATABASE_URL includes a dollar sign in the password section.

The following works with prisma cli but not prisma client:
`"mongodb+srv://user:some$password@host/db"`

The recommendation is to escape the dollar sign using a backslash.
The following works with prisma client but not prisma cli (correct behaviour): 
`"mongodb+srv://user:some\$password@host/db"`

I also tried using single quotes without escaping dollar sign but the behaviour is exactly the same as double quotes. In my opinion, the prisma cli should be updated to use the correct behaviour. This might be an underlying bug in npx itself in the way it reads and forwards environment variables to the prisma cli but I am unable to verify the same.

Right now, my only solution is to run the prisma cli without the escape character in the environment variable value. And then add the escape character before reloading the app.

### How to reproduce

1. Create a user on your choice of database with a dollar sign in the password.
2. In the `.env` file, keep the dollar sign unescaped
3. Try running a prisma cli command that interacts with the database.
4. You will not get an error.
5. Try running a query from prisma client.
6. You will get authentication failure.

### Expected behavior

Prisma CLI should follow the same behaviour as Prisma Client i.e., allow escape characters:
`"mongodb+srv://user:some\$password@host/db"`

### Prisma information

```prisma
// Not relevant
```

```ts
// Not relevant
```

### Environment & setup

- OS: WSL 2
- Database: MongoDB
- Node.js version: v18.12.1
- NPM Version: 8.19.2

### Prisma Version

```
4.7.1
```

**Comment by janpio** [comment]
In your application the environment variables were loaded via Next.js logic?

**Comment by arunlodhi** [comment]
Yes, in my application, the prisma client is initialized within a NextJS API route and NextJS follows the convention of expanding environment variables. Hence I need to escape the dollar sign while loading the app and then remove the escape character while running prisma cli.

**Comment by janpio** [comment]
Thanks, I think that realization explains quite a few problems that have been reported around env vars over time - but as far as I know we never connected these dots properly.

**Comment by arunlodhi** [comment]
Got it. I think the issue can be resolved by adding a flag to prisma cli instructing it to expand environment variables. I just checked locally using the method explained at:

https://github.com/motdotla/dotenv-expand

.env:
```env
DATABASE_URL="mongodb+srv://user:some\$password@host/db"
```

js:
```js
var dotenv = require('dotenv');
var dotenvExpand = require('dotenv-expand');

var myEnv = dotenv.config();
dotenvExpand.expand(myEnv);

console.log(process.env.DATABASE_URL);   // mongodb+srv://user:some$password@host/db
```
The above method gives me the correct DATABASE_URL consistent with prisma client for my use case. To enable this optional behaviour, we could have a flag like:
```
npx prisma db push --expand-env-vars
```
or this could also be moved to a cli config file `prisma-cli.config.js`.

Let me know your thoughts on this.
